### PR TITLE
[SYCLomatic] Fix <oneapi/dpl/execution> and <oneapi/dpl/algorithm> are not inserted

### PR DESCRIPTION
**Pull request by yihwang-nv**
Signed-off-by: Wang, Yihan <yihan.wang@intel.com>

### CODE REVIEW

File: clang/lib/DPCT/AnalysisInfo.h [hunk 1]
```diff
@@ -2456,6 +2374,7 @@ class DpctGlobalInfo {
       PriorityReplInfoMap;
   static std::unordered_map<std::string, bool> ExcludePath;
   static std::map<std::string, clang::tooling::OptionInfo> CurrentOptMap;
+  static DpctFileInfo *MainFile;
```
Comment (yihwang-nv) [inline clang/lib/DPCT/AnalysisInfo.h]: use shared_ptr

File: clang/lib/DPCT/AnalysisInfo.cpp [hunk 2]
```diff
@@ -365,6 +367,12 @@ void DpctFileInfo::buildLinesInfo() {
   if (std::error_code ec = FE.getError())
     return;
   auto FID = SM.getOrCreateFileID(FE.get(), SrcMgr::C_User);
+  FileID = FID;
+
+  if (SM.isMainFile(**FE)) {
+    DpctGlobalInfo::getInstance().setMainFile(this);
```
Comment (yihwang-nv) [inline clang/lib/DPCT/AnalysisInfo.cpp]: use BeginSourceFileAction

**Review by AndyCHHuang — APPROVED** [review]
LGTM

**Review by ziranzha — APPROVED** [review]
LGTM

**Pull request merged**